esp/esp/program/models/class_.py

```diff
@@ -1002,6 +1005,11 @@ def cancel(self, email_students=True, email_teachers=True, include_lottery_stude
                 else:
                     send_mail(ssi_email_title, msgtext, from_email, to_email)
 
+        if text_students and self.parent_program.hasModule('GroupTextModule') and GroupTextModule.is_configured():
+            if self.students(student_verbs).distinct().count() > 0:
+                msgtext = render_to_string('texts/class_cancellation.txt', context)
+                students_to_text = PersistentQueryFilter.create_from_Q(ESPUser, Q(id__in=[x.id for x in self.students(student_verbs)]))
```

Comment:
I think you may have accidentally also the line that actually sends the messages!